// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

// Definitions for common cryptographic enum types.
syntax = "proto3";

package google.cloud.crypto.tink;

option java_package = "com.google.cloud.crypto.tink";
option java_outer_classname = "CommonProto";

enum EllipticCurveType {
  UNKNOWN_CURVE = 0;
  NIST_P224 = 1;
  NIST_P256 = 2;
  NIST_P384 = 3;
  NIST_P521 = 4;
};

enum EcPointFormat {
  UNKNOWN_FORMAT = 0;
  UNCOMPRESSED = 1;
  COMPRESSED = 2;
}

enum HashType {
  UNKNOWN_HASH = 0;
  SHA1 = 1; // Using SHA1 for digital signature is deprecated but HMAC-SHA1 is fine.
  SHA224 = 2;
  SHA256 = 3;
  SHA512 = 4;
};
